Terminal.Gui/Views/TextView.cs

```diff
@@ -30,61 +94,81 @@ public bool CloseFile ()
 				throw new ArgumentNullException (nameof (FilePath));
 
 			FilePath = null;
-			_lines = new List<List<Rune>> ();
+			_lines = new List<List<RuneCell>> ();
 			return true;
 		}
 
-		// Turns the string into runes, this does not split the 
+		// Turns the string into cells, this does not split the 
 		// contents on a newline if it is present.
-		internal static List<Rune> ToRunes (string str)
+		internal static List<RuneCell> StringToRuneCells (string str, ColorScheme? colorScheme = null)
 		{
-			List<Rune> runes = new List<Rune> ();
-			foreach (var x in str.ToRunes ()) {
-				runes.Add (x);
+			List<RuneCell> cells = new List<RuneCell> ();
+			foreach (var rune in str.ToRunes ()) {
+				cells.Add (new RuneCell { Rune = rune, ColorScheme = colorScheme });
 			}
-			return runes;
+			return cells;
 		}
 
-		// Splits a string into a List that contains a List<Rune> for each line
-		public static List<List<Rune>> StringToRunes (string content)
+		internal static List<RuneCell> ToRuneCells (IEnumerable<Rune> runes, ColorScheme? colorScheme = null)
 		{
-			var lines = new List<List<Rune>> ();
+			List<RuneCell> cells = new List<RuneCell> ();
+			foreach (var rune in runes) {
+				cells.Add (new RuneCell { Rune = rune, ColorScheme = colorScheme });
+			}
+			return cells;
+		}
+
+		public static List<List<RuneCell>> ToRuneCells (List<RuneCell> cells)
```

Comment:
This API is called once, from what I can tell. Please don't expose `public` APIs that are simple helpers!

If I'm reading this code right, a `List<List<RuneCell>>` is "a list of Lines, where each line is defined as a `List<RuneCell>`".

To make all of this code more readable and understandable, would it make sense to introduce a data structure/class `Line`:

```cs
		/// <summary>
		/// Represents a line of text in the <see cref="TextModel"/>.
		/// </summary>
		public class Line : List<RuneCell> {
		}
```

Reply:
> This API is called once, from what I can tell. Please don't expose public APIs that are simple helpers!

Confirmed, but that was already public before I rename it to `ToRuneCells`. I'll make it private then.

> If I'm reading this code right, a List<List<RuneCell>> is "a list of Lines, where each line is defined as a List<RuneCell>".

Right.

> To make all of this code more readable and understandable, would it make sense to introduce a data structure/class Line:

Well, I think this is a contradiction. If you don't want expose nothing that does not is used on others class, why you want to expose the Line. In `TextModel` the `_lines` is a private member field. His value is get with methods, like `GetLine`, `GetAllLines`. All these code are intensively tied to `TextView` and `TextField` and adding more complexity to the private `TextModel`, in this case, I think is not worth.

Reply:
Done via https://github.com/gui-cs/Terminal.Gui/pull/2682/commits/02c3108b4e414d3b77aade6c66c6769ca1305925.